src/entities/supportRequest.ts

```diff
@@ -99,16 +99,25 @@ export class SupportRequest extends BaseEntity {
     }
   }
 
-  static async getNextSupportRequest(): Promise<SupportRequest> {
+  static async getNextSupportRequest(supportType: SupportRequestType): Promise<SupportRequest> {
     let retries = 5;
 
     do {
-      const nextRequest = await SupportRequest.createQueryBuilder('supportRequests')
-        .orderBy('"supportRequests"."createdAt"', 'ASC')
-        .where({
-          status: SupportRequestStatus.Pending,
-        })
-        .getOne();
+      const nextRequest =
+        supportType === null
```

Comment:
We actually don't need to evaluate the type since a value will be **_truthy_** here; basically "is it or isn't it"
```suggestion
        supportType
```